### DISCUSSION TITLE
Feature Request - Preset last values in new entry form

### DISCUSSION THREAD

**Discussion by francwalter** [kimai/kimai#3933] [General]
Hallo

I have mostly one customer. When I add an entry I often just add the complete time I spent, I add my notes (what I worked exactly) and I dont start the timer when I begin (to end it later when finished). This is my mostly used workflow.

Now when I add a time I use either the button for re-start (a circle arrow) or the Plus-Button for a new time slip.
When I use the Plus-Button I have to fill in all values (Customer, Project etc), here I would love, if the last used values were pre-filled.
Not the notes, though.
When I use the Re-Start-Button, I can choose the last customer and the values are prefilled, that ist good, but I have to stop the timer which is a bit unhandy as I may (and did) forget it (by the way, there is no hover help on any of those buttons).

So it would be a nice easier making UX if the values could be (maybe with a setting) pre-filled using the Plus Sign to add times.
Is this possible?
Thank.
frank

Here an explanation what I mean with Plus-Button and Re-Start-Button:
![image](https://user-images.githubusercontent.com/927269/225587996-53dfa0b8-c353-4d9c-bac7-055c78806327.png)

**Comment by kevinpapst** [comment]
Use the "Create copy" action of each timesheet?

**Reply by francwalter** [reply]
> Use the "Create copy" action of each timesheet?

Sorry that I insist, but is it a proposition which I maybe just oversaw?
Is there maybe a button "Create copy" somewhere? But where?
There is only the Re-Start button, as I wrote already, but I dont want to start, I want to copy without running.
Could you explain for me as a "Dummie-user"?
Thank!

**Reply by kevinpapst** [reply]
<img width="214" alt="Bildschirm­foto 2023-03-22 um 16 48 46" src="https://user-images.githubusercontent.com/533162/226961732-7aa4f597-1797-40b5-b95e-a150e6e43f5d.png">

**Reply by francwalter** [reply]
Ahhh! OK, now I know.
Look:

![grafik](https://user-images.githubusercontent.com/927269/226967274-232ab5f4-6074-4d15-9d28-03b02cf1f230.png)

The too much headers made the DropDown Button disappear 😄
But there is no scrolling possible. The collumns are not sizeable neither. No way to see the DropDown Button if the list is spawned to wide by the headers. 
I unchecked the "Schlagworte" and now I can see (and use) it 😄 

I will try to use that "Create copy" and hope it is faster than the other way.

Thanks.

**Comment by francwalter** [comment]
Where can I find that?
It is unmentioned in the [online help](https://www.kimai.org/documentation/timesheet.html). There is only the re-start (as I wrote in my beginning already) but no copy of timesheets.
In the timesheet there is nothing like that.
Maybe my version is too old? 
I have Kimai2 v1.30.2

Could be easy to implement: if the user setting "autofill last entries" is set, then look in the db about the last entry of the user and pre-fill the values with these last values.
Yes, I know: I should do it and then PR it. But it would take me weeks till I had it programmed and then in the end it would be too ugly coded and no PR would be accepted :smirk:
